pkg/scheduler/internal/cache/cache.go

```diff
@@ -521,7 +539,6 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 		if err := cache.updatePod(oldPod, newPod); err != nil {
```

Comment:
isn't this now losing information about the podState if there was no change of UID?

Reply:
Yes, but since `!cache.assumedPods.Has(key)` I think there is no other information except the pod.

Reply:
agree.

Reply:
Sounds good, thanks for checking.

With that simplification, everything looks pretty consistent and maintainable now.